base/src/main/java/com/austindroids/austinfeedsme/data/firebase/FirebaseEventsDataSource.java

```diff
@@ -31,19 +30,24 @@ public FirebaseEventsDataSource(CollectionReference fireBase) {
     }
 
     @Override
-    public void getEvents(final LoadEventsCallback callback, boolean onlyFood) {
-        final List<Event> events = new ArrayList<>();
-        Query eventsQuery = collectionReference
-                .whereGreaterThan("time", new Date().getTime())
-                .orderBy("time");
-        if (onlyFood) {
-            eventsQuery.whereEqualTo("food", true);
-        }
+    public void getEvents(final LoadEventsCallback callback) {
+        getEvents(callback, false, false);
+    }
 
+    @Override
+    public void saveEvent(Event eventToSave, SaveEventCallback callback) {
+        collectionReference.add(eventToSave);
+        callback.onEventSaved(true);
+    }
+
+    @Override
+    public void getEvents(LoadEventsCallback callback, boolean futureEvents, boolean foodOnly) {
+        Query eventsQuery = getEventsQuery(futureEvents, foodOnly);
         eventsQuery.get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
             @Override
             public void onComplete(@NonNull Task<QuerySnapshot> task) {
                 if (task.isSuccessful()) {
+                    List<Event> events = new ArrayList<>();
```

Comment:
You can get some small performance optimization by getting `task.getResult()` into a variable, then initialize the arraylist with the size: `new ArrayList<>(result.size())`